scram-tools.file/tools/tensorflow-xla-runtime/tensorflow-xla-runtime.xml

```diff
@@ -1,9 +1,11 @@
 <tool name="tensorflow-xla-runtime" version="@TOOL_VERSION@">
   <client>
     <environment name="TENSORFLOW_XLA_RUNTIME_BASE" default="@TOOL_ROOT@"/>
-    <environment name="LIBDIR" default="$TENSORFLOW_XLA_RUNTIME_BASE/lib/archive"/>
+    <environment name="LIBDIR" default="$TENSORFLOW_XLA_RUNTIME_BASE/lib"/>
   </client>
-  <lib name="tf_xla_runtime-static"/>
+  <lib name="tf_xla_runtime"/>
+  <flags LDFLAGS="-Wl,--unresolved-symbols=ignore-in-shared-libs"/>
```

Comment:
@riga , we do not need these ld flags any more .. right?

Reply:
ah ok, I now have read your https://github.com/cms-sw/cmsdist/pull/9093#discussion_r1538034992 . I will check what else we can do to avoid the missing symbols. We should not add this flag here otherwise scram will not fail for missing symbols for every cmssw shared lib which depend on tensorflow-xml-runtime  (directly or in-directly)

Reply:
Yeah, this already felt like a dangerous thing to do. Do you think there is any other option besides a) fetching the missing source files to `tensorflow/tsl`, or b) installing the [full tsl](https://github.com/google/tsl) as a dependency (still seems a bit like work in progress)?

Reply:
@riga , all the missing TSL symbols are coming from https://github.com/tensorflow/tensorflow/blob/v2.12.1/tensorflow/compiler/xla/service/cpu/runtime_fork_join.cc e.g various calls macros calls `VLOG, CHECK_*` and `tsl::BlockingCounter`. If you think we can run without these missing symbols ( e.g. we never call these) so why not just then drop the `runtime_fork_join.cc ` from the compilation?

Reply:
> If you think we can run without these missing symbols ( e.g. we never call these) so why not just then drop the runtime_fork_join.cc  from the compilation?

Ok, so when we used the static lib before, the corresponding object files were likely dropped already, makes sense. And since we will potentially never use eigen thread pools for inference, we can skip it right away. Then I guess we are lucky that the tsl dependence only affects parts the xla runtime source that we anyway don't need (well, at least for now).

We just verified that everything still works 👍